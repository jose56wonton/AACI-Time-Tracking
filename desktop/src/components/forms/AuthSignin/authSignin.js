import React from 'react';
import PropTypes from 'prop-types';

import { Typography, Button } from '@material-ui/core';
import { withStyles } from '@material-ui/core/styles';
import { Field, Form } from 'formik';

import TextField from 'components/inputs/TextField';
import Password from 'components/inputs/Password';
import styles from './styles';

const AuthSigninForm = props => {
  const { classes, isSubmitting, errors } = props;

  return (
    <div className={classes.hero}>
      <Form className={classes.heroContent}>
        <Typography variant="h6">Select Network</Typography>
        <Field component={TextField} name="username" label="Network" helper="none"/>
        <Field component={Password} name="password" label="Password" helper="none" />
        <Button
          className={classes.button}
          type="submit"
          color="primary"
          variant="contained"
          disabled={isSubmitting}
        >
          Submit
        </Button>
      </Form>
      <div className={classes.error}>
        <Typography color="error">
          {errors.submit}
        </Typography>
      </div>
    </div>
  );
};

AuthSigninForm.propTypes = {
  classes: PropTypes.object.isRequired,
  errors: PropTypes.object.isRequired,
  onSubmit: PropTypes.func,
  isSubmitting: PropTypes.bool
};

export default withStyles(styles)(AuthSigninForm);
