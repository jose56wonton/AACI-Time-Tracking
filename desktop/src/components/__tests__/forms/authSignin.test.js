import React from 'react';
import { shallow } from 'enzyme';

import { AuthSignin } from 'components/forms/AuthSignin/authSignin';
import AuthSigninHOC from 'components/forms/AuthSignin';

const props =  {  
  classes: {},
  isSubmitting: true,
  errors: {}
};

const setup = overRides => {  
  return shallow(<AuthSignin {...props} {...overRides}/>);    
};

const setupHOC = overRides => {
  return shallow(<AuthSigninHOC {...props} {...overRides}/>);
};

describe.skip('Auth Signin Component', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  it('should render correctly', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();    
  });
  it('should render correctly withStyles', () => {
    const wrapper = setupHOC();
    expect(wrapper).toMatchSnapshot();
  });
});