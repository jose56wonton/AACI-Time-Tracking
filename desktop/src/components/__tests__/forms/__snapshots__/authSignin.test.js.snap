// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth Signin Component should render correctly 1`] = `
<div>
  <Form>
    <WithStyles(Typography)
      variant="h6"
    >
      Select Network
    </WithStyles(Typography)>
    <C
      component={[Function]}
      helper="none"
      label="Network"
      name="username"
    />
    <C
      component={[Function]}
      helper="none"
      label="Password"
      name="password"
    />
    <WithStyles(Button)
      color="primary"
      disabled={true}
      type="submit"
      variant="contained"
    >
      Submit
    </WithStyles(Button)>
  </Form>
  <div>
    <WithStyles(Typography)
      color="error"
    />
  </div>
</div>
`;

exports[`Auth Signin Component should render correctly withStyles 1`] = `
<Component
  classes={
    Object {
      "button": "Component-button-4",
      "error": "Component-error-3",
      "hero": "Component-hero-1",
      "heroContent": "Component-heroContent-2",
    }
  }
  errors={Object {}}
  isSubmitting={true}
/>
`;
